src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPool.cs

```diff
@@ -109,4 +226,46 @@ protected override void Dispose(bool disposing)
             }
         }
     }
+
+    // Used for testing
+    public int BlockCount() => _blocks.Count;
+
+    private sealed class NoopMeterFactory : IMeterFactory
+    {
+        public static NoopMeterFactory Instance { get; } = new();
+
+        public Meter Create(MeterOptions options) => new Meter(options);
+
+        public void Dispose()
+        {
+        }
+    }
+
+    // https://github.com/dotnet/runtime/blob/db681fb307d754c3746ffb40e0634e4c4e0caa9e/docs/design/features/ScalableApproximateCounting.md
+    static void ScalableCount(ref uint counter)
```

Comment:
@AndyAyersMS !

Reply:
But it's `#if`'d out.... is this here just in case you find you have false sharing?

Reply:
I'm still testing perf of interlocked vs. ScalableApproximateCounting for our scenarios.

Reply:
I did some perf comparisons and didn't notice any issues with interlocked, so going with that approach for now.

Reply:
Yeah I would not expect to see much false sharing for isolated integer fields, since two such counters are unlikely to be able to share the same cache line.

Reply:
I just really wanted to use ScalableApproximateCounting for something 😆 
Next time™️